src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -14861,6 +14923,594 @@ public void Main(string key)
             c.VerifyDiagnostics();
         }
 
+        [Fact]
+        [WorkItem(29295, "https://github.com/dotnet/roslyn/issues/29295")]
+        public void CatchException()
+        {
+            var c = CreateCompilation(@"
+class C
+{
+    void M()
+    {
+        try
+        {
+            System.Console.WriteLine();
+        }
+        catch (System.Exception e)
+        {
+            e.ToString();
+        }
+    }
+}
+", options: WithNonNullTypesTrue());
+
+            c.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(29295, "https://github.com/dotnet/roslyn/issues/29295")]
+        public void CatchException_WithWhenIsOperator()
+        {
+            var c = CreateCompilation(@"
+class C
+{
+    void M()
+    {
+        try
+        {
+            System.Console.WriteLine();
+        }
+        catch (System.Exception e) when (!(e is System.ArgumentException))
+        {
+            e.ToString();
+        }
+    }
+}
+", options: WithNonNullTypesTrue());
+
+            c.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(26739, "https://github.com/dotnet/roslyn/issues/26739")]
+        public void RefParameter_TopLevelNullability()
+        {
+            var c = CreateCompilation(@"
+class C
+{
+    public void M()
+    {
+        string? s = string.Empty;
+        M(ref s);
+        s.ToString(); // 1
+
+        string s2 = string.Empty;
```

Comment:
How is this second part different from the added warning in `PassingParameters_01`, other than the extra call? If both of these aren't different, consider removing this test and adding the call to `PassingParameters_01`.

Reply:
Or otherwise consolidating the tests.

---
In reply to: [258176215](https://github.com/dotnet/roslyn/pull/33447#discussion_r258176215) [](ancestors = 258176215)

Reply:
I'll remove `MethodWithRefNullableParameter` which is indeed subsumed by this test. Thanks
As for `PassingParameters_01`, I'd prefer to leave that as-is (a little bit of redundancy, but easier to follow in my opinion).


---
In reply to: [258176369](https://github.com/dotnet/roslyn/pull/33447#discussion_r258176369) [](ancestors = 258176369,258176215)